Ak96/refactor contributor script again

aaruni96 opened the pull request.

aaruni96 committed:
Step1: easy linter complaints

aaruni96 committed:
Step2: move function definitions together

aaruni96 committed:
step3: get rid of internal names

aaruni96 committed:
refactor norm_name()

aaruni96 committed:
refactor lookup_user

aaruni96 committed:
Improvement: more natural sorting of people

[skip ci]

fingolfin commented on etc/update-contributors.py:
```diff
@@ -48,72 +52,89 @@
+    "any repo.\n"
+    "# Retired people only have repo information if repo information about them was known when "
+    "they were\n"
+    "# active (or manually added) by a maintainer.\n\n"
```
This looks really weird. How about using a triple quoted string, and just wrap the lines here as they'll be wrapped in the generated file?

fingolfin commented on etc/update-contributors.py:
```diff
@@ -48,72 +52,89 @@
 
 def process_log_into_aggregate(res: str, repo: str) -> None:
-    global summarystring, aggregate  #, email_owner, name_owner
+    global summarystring, aggregate  # email_owner, name_owner
```
```suggestion
    global summarystring, aggregate
```

fingolfin commented on etc/update-contributors.py:
```diff
@@ -48,72 +52,89 @@
 
-def _norm_name(s: str) -> str:
-    return " ".join((s or "").split()).casefold()
+def find_author_github_nick(email: str, repos: list[str]) -> str:
```
Maybe rename to `find_github_username`? This is not a "nick" or "nickname", after all.

fingolfin commented on etc/update_contributors.py:
```diff
@@ -48,72 +52,89 @@
-    return " ".join((s or "").split()).casefold()
+def find_author_github_nick(email: str, repos: list[str]) -> str:
+    if not email:
+        return ""
```
This used to return `None`. The value is then passed on to `lookup_user` which does not seem ready for this change? (I also find `None` logically cleaner, but that's a matter of taste *shrug*)

aaruni96 replied:
In python, `lookup_user` does not need to be refactored to accommodate an empty string. `if None` and `if ""` behave the same way (they are both "falsy").

I made this change because to me it looks "cleaner" that a function always returns the same type. But beyond this ("looks cleaner to my eyes"), there's no strong reason, and I can add back the `None`.

fingolfin replied:
as long as it works, fine by ,e

HereAround commented:
@aaruni96 Just a pointer that this will have conflicts with https://github.com/oscar-system/oscar-website/pull/754 (once merged). What is the status of this draft?

aaruni96 committed:
some more "fixes"

aaruni96 commented:
Did a rebase against `gh-pages`.

The linter still complains about lots of things, so, at minimum, those are still todo (either fix them, or decide that we ignore them)

```
etc/update_contributors.py:157:13: W0511: todo log a warning here (fixme)
etc/update_contributors.py:1:0: C0114: Missing module docstring (missing-module-docstring)
etc/update_contributors.py:100:0: C0116: Missing function or method docstring (missing-function-docstring)
etc/update_contributors.py:110:0: C0116: Missing function or method docstring (missing-function-docstring)
etc/update_contributors.py:110:34: W0621: Redefining name 'email' from outer scope (line 309) (redefined-outer-name)
etc/update_contributors.py:110:46: W0621: Redefining name 'name' from outer scope (line 309) (redefined-outer-name)
etc/update_contributors.py:112:8: W0621: Redefining name 'user' from outer scope (line 311) (redefined-outer-name)
etc/update_contributors.py:120:0: C0116: Missing function or method docstring (missing-function-docstring)
etc/update_contributors.py:120:12: W0621: Redefining name 'repo_dir' from outer scope (line 288) (redefined-outer-name)
etc/update_contributors.py:132:0: C0116: Missing function or method docstring (missing-function-docstring)
etc/update_contributors.py:139:0: C0116: Missing function or method docstring (missing-function-docstring)
etc/update_contributors.py:139:25: W0621: Redefining name 'email' from outer scope (line 309) (redefined-outer-name)
etc/update_contributors.py:139:37: W0621: Redefining name 'repos' from outer scope (line 309) (redefined-outer-name)
etc/update_contributors.py:155:15: W3101: Missing timeout argument for method 'requests.get' can cause your program to hang indefinitely (missing-timeout)
etc/update_contributors.py:165:0: C0116: Missing function or method docstring (missing-function-docstring)
etc/update_contributors.py:165:25: W0621: Redefining name 'name' from outer scope (line 309) (redefined-outer-name)
etc/update_contributors.py:165:36: W0621: Redefining name 'email' from outer scope (line 309) (redefined-outer-name)
etc/update_contributors.py:165:48: W0621: Redefining name 'repos' from outer scope (line 309) (redefined-outer-name)
etc/update_contributors.py:185:0: C0116: Missing function or method docstring (missing-function-docstring)
etc/update_contributors.py:185:41: W0621: Redefining name 'repo' from outer scope (line 286) (redefined-outer-name)
etc/update_contributors.py:204:8: W0621: Redefining name 'name' from outer scope (line 309) (redefined-outer-name)
etc/update_contributors.py:204:14: W0621: Redefining name 'email' from outer scope (line 309) (redefined-outer-name)
etc/update_contributors.py:222:12: W0621: Redefining name 'key' from outer scope (line 308) (redefined-outer-name)
etc/update_contributors.py:226:8: W0621: Redefining name 'rec' from outer scope (line 308) (redefined-outer-name)
etc/update_contributors.py:186:4: W0603: Using the global statement (global-statement)
```

aaruni96 committed:
Minor cosmetic changes

HereAround committed:
Fix some pylint issues

HereAround committed:
Wrap lines as suggested by pylint

HereAround committed:
Address timeout and todo

HereAround committed:
Address the final two pylint issues

HereAround commented:
> Did a rebase against `gh-pages`.
> 
> The linter still complains about lots of things, so, at minimum, those are still todo (either fix them, or decide that we ignore them)
> 
> ```
> etc/update_contributors.py:157:13: W0511: todo log a warning here (fixme)
> etc/update_contributors.py:1:0: C0114: Missing module docstring (missing-module-docstring)
> etc/update_contributors.py:100:0: C0116: Missing function or method docstring (missing-function-docstring)
> etc/update_contributors.py:110:0: C0116: Missing function or method docstring (missing-function-docstring)
> etc/update_contributors.py:110:34: W0621: Redefining name 'email' from outer scope (line 309) (redefined-outer-name)
> etc/update_contributors.py:110:46: W0621: Redefining name 'name' from outer scope (line 309) (redefined-outer-name)
> etc/update_contributors.py:112:8: W0621: Redefining name 'user' from outer scope (line 311) (redefined-outer-name)
> etc/update_contributors.py:120:0: C0116: Missing function or method docstring (missing-function-docstring)
> etc/update_contributors.py:120:12: W0621: Redefining name 'repo_dir' from outer scope (line 288) (redefined-outer-name)
> etc/update_contributors.py:132:0: C0116: Missing function or method docstring (missing-function-docstring)
> etc/update_contributors.py:139:0: C0116: Missing function or method docstring (missing-function-docstring)
> etc/update_contributors.py:139:25: W0621: Redefining name 'email' from outer scope (line 309) (redefined-outer-name)
> etc/update_contributors.py:139:37: W0621: Redefining name 'repos' from outer scope (line 309) (redefined-outer-name)
> etc/update_contributors.py:155:15: W3101: Missing timeout argument for method 'requests.get' can cause your program to hang indefinitely (missing-timeout)
> etc/update_contributors.py:165:0: C0116: Missing function or method docstring (missing-function-docstring)
> etc/update_contributors.py:165:25: W0621: Redefining name 'name' from outer scope (line 309) (redefined-outer-name)
> etc/update_contributors.py:165:36: W0621: Redefining name 'email' from outer scope (line 309) (redefined-outer-name)
> etc/update_contributors.py:165:48: W0621: Redefining name 'repos' from outer scope (line 309) (redefined-outer-name)
> etc/update_contributors.py:185:0: C0116: Missing function or method docstring (missing-function-docstring)
> etc/update_contributors.py:185:41: W0621: Redefining name 'repo' from outer scope (line 286) (redefined-outer-name)
> etc/update_contributors.py:204:8: W0621: Redefining name 'name' from outer scope (line 309) (redefined-outer-name)
> etc/update_contributors.py:204:14: W0621: Redefining name 'email' from outer scope (line 309) (redefined-outer-name)
> etc/update_contributors.py:222:12: W0621: Redefining name 'key' from outer scope (line 308) (redefined-outer-name)
> etc/update_contributors.py:226:8: W0621: Redefining name 'rec' from outer scope (line 308) (redefined-outer-name)
> etc/update_contributors.py:186:4: W0603: Using the global statement (global-statement)
> ```

I just pushed changes to address the above points.

Note that the script now finds a newly retired contributor:

 Newly retired contributors : 1 | ['Felix Röhrich']
 
 We should conduct tests to verify that the script works as intended.

aaruni96 committed:
Some fixes:

- turn summary_string into lower case
- Add main() to get rid of global, use if-name-main convention
- remove aggregate_key, as it was not doing anything

HereAround reviewed (approved):
Thank you very much @aaruni96 . LGTM and will merge now.
